### ISSUE TITLE
Class 'App\Http\Controllers\Toastr' not found

### ISSUE THREAD

**Issue by petrovitch** [nilsenj/toastr-5.1-laravel#1]
Not recognizing the Facade. Am I missing something about "use" some method in the controller?

css/js referenced in master layout

View:

```
    {!! Toastr::render() !!}
```

Controller:

```
Toastr::info('This is a test');
```

Class 'App\Http\Controllers\Toastr' not found

**Comment by nilsenj** [comment]
first include         'Toastr' => 'nilsenj\Toastr\Facades\Toastr', alias into app.php in the config folder
then in controller or somewhere else 
use nilsenj\Toastr\Facades\Toastr; or just simply use Toastr;
then it should work fine!
by the way if you think about some extra functionality ask me and I'll try to do something else.
if you for some reasons have a problem
run 
composer dump-autoload

**Comment by petrovitch** [comment]
I don't know what is going wrong, but I can't make it work.  :-(  This
should be so simple, yet it is not working for me.  I can post this simple
project on github if it would help.

By the way, thanks for your reply.

"require": {
    "php": ">=5.5.9",
    "laravel/framework": "5.1._",
    "laravelcollective/html": "5.1._",
    "guzzlehttp/guzzle": "~5.3|~6.0",
    "zizaco/entrust": "dev-laravel-5",
    "tecnickcom/tcpdf": "6.0.*",
    "maatwebsite/excel": "~2.0",
    "nesbot/carbon": "^1.20",
    "rap2hpoutre/laravel-log-viewer": "^0.4.2",
    "owen-it/laravel-auditing": "^2.0",
-    "nilsenj/toastr-5.1-laravel": "dev-master"*

'providers' => [

```
/*
 * Laravel Framework Service Providers...
 */
Illuminate\Foundation\Providers\ArtisanServiceProvider::class,
Illuminate\Auth\AuthServiceProvider::class,
Illuminate\Broadcasting\BroadcastServiceProvider::class,
Illuminate\Bus\BusServiceProvider::class,
Illuminate\Cache\CacheServiceProvider::class,
Illuminate\Foundation\Providers\ConsoleSupportServiceProvider::class,
Illuminate\Routing\ControllerServiceProvider::class,
Illuminate\Cookie\CookieServiceProvider::class,
Illuminate\Database\DatabaseServiceProvider::class,
Illuminate\Encryption\EncryptionServiceProvider::class,
Illuminate\Filesystem\FilesystemServiceProvider::class,
Illuminate\Foundation\Providers\FoundationServiceProvider::class,
Illuminate\Hashing\HashServiceProvider::class,
Illuminate\Mail\MailServiceProvider::class,
Illuminate\Pagination\PaginationServiceProvider::class,
Illuminate\Pipeline\PipelineServiceProvider::class,
Illuminate\Queue\QueueServiceProvider::class,
Illuminate\Redis\RedisServiceProvider::class,
Illuminate\Auth\Passwords\PasswordResetServiceProvider::class,
Illuminate\Session\SessionServiceProvider::class,
Illuminate\Translation\TranslationServiceProvider::class,
Illuminate\Validation\ValidationServiceProvider::class,
Illuminate\View\ViewServiceProvider::class,

/*
 * Application Service Providers...
 */
App\Providers\AppServiceProvider::class,
App\Providers\AuthServiceProvider::class,
App\Providers\EventServiceProvider::class,
App\Providers\RouteServiceProvider::class,
Collective\Html\HtmlServiceProvider::class,
'Zizaco\Entrust\EntrustServiceProvider',
'Maatwebsite\Excel\ExcelServiceProvider',
Rap2hpoutre\LaravelLogViewer\LaravelLogViewerServiceProvider::class,
OwenIt\Auditing\AuditingServiceProvider::class,
```
-    'nilsenj\Toastr\ToastrServiceProvider',*],

/*
|--------------------------------------------------------------------------
| Class Aliases
|--------------------------------------------------------------------------
|
| This array of class aliases will be registered when this application
| is started. However, feel free to register as many as you wish as
| the aliases are "lazy" loaded so they don't hinder performance.
|
*/

'aliases' => [

```
'App'       => Illuminate\Support\Facades\App::class,
'Artisan'   => Illuminate\Support\Facades\Artisan::class,
'Auth'      => Illuminate\Support\Facades\Auth::class,
'Blade'     => Illuminate\Support\Facades\Blade::class,
'Bus'       => Illuminate\Support\Facades\Bus::class,
'Cache'     => Illuminate\Support\Facades\Cache::class,
'Config'    => Illuminate\Support\Facades\Config::class,
'Cookie'    => Illuminate\Support\Facades\Cookie::class,
'Crypt'     => Illuminate\Support\Facades\Crypt::class,
'DB'        => Illuminate\Support\Facades\DB::class,
'Eloquent'  => Illuminate\Database\Eloquent\Model::class,
'Event'     => Illuminate\Support\Facades\Event::class,
'File'      => Illuminate\Support\Facades\File::class,
'Gate'      => Illuminate\Support\Facades\Gate::class,
'Hash'      => Illuminate\Support\Facades\Hash::class,
'Input'     => Illuminate\Support\Facades\Input::class,
'Inspiring' => Illuminate\Foundation\Inspiring::class,
'Lang'      => Illuminate\Support\Facades\Lang::class,
'Log'       => Illuminate\Support\Facades\Log::class,
'Mail'      => Illuminate\Support\Facades\Mail::class,
'Password'  => Illuminate\Support\Facades\Password::class,
'Queue'     => Illuminate\Support\Facades\Queue::class,
'Redirect'  => Illuminate\Support\Facades\Redirect::class,
'Redis'     => Illuminate\Support\Facades\Redis::class,
'Request'   => Illuminate\Support\Facades\Request::class,
'Response'  => Illuminate\Support\Facades\Response::class,
'Route'     => Illuminate\Support\Facades\Route::class,
'Schema'    => Illuminate\Support\Facades\Schema::class,
'Session'   => Illuminate\Support\Facades\Session::class,
'Storage'   => Illuminate\Support\Facades\Storage::class,
'URL'       => Illuminate\Support\Facades\URL::class,
'Validator' => Illuminate\Support\Facades\Validator::class,
'View'      => Illuminate\Support\Facades\View::class,
'Form'      => Collective\Html\FormFacade::class,
'Html'      => Collective\Html\HtmlFacade::class,
'Entrust'   => 'Zizaco\Entrust\EntrustFacade',
'Excel'     => 'Maatwebsite\Excel\Facades\Excel',
```
-    'Toastr'    => 'nilsenj\Toastr\Facades\Toastr',*],

<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Http\Requests;
use App\Http\Controllers\Controller;
use Illuminate\Support\Facades\DB;

_use Toastr;_
class PagesController extends Controller
{
    public function about()
    {
-        Toastr::warning('This is a test.');\*        return view('about');
  }

<!DOCTYPE html>
<html>
<head>
    <title> @yield('title') </title>
    <link rel="stylesheet" type="text/css" href="{!! asset('css/app.css') !!}">
    {{--<link rel="stylesheet" type="text/css" href="{!!
asset('css/bootstrap.min.css') !!}">--}}
    {{--<link rel="stylesheet" type="text/css" href="{!!
asset('css/bootstrap-theme.min.css') !!}">--}}

```
{{--Material Design Theme--}}
<link rel="stylesheet" type="text/css" href="{!!
```

asset('css/roboto.min.css') !!}">
    <link rel="stylesheet" type="text/css" href="{!!
asset('css/material.min.css') !!}">
    <link rel="stylesheet" type="text/css" href="{!!
asset('css/ripples.min.css') !!}">

```
<link href="css/toastr.css" rel="stylesheet" media="screen"></head>
```

<body>

_{!! Toastr::render() !!}_
@include('partials.navbar')

@yield('content')

<script src="{!! asset('js/jquery-1.11.3.min.js') !!}"></script>

<script src="{!! asset('js/bootstrap.min.js') !!}"></script>

{{--Material Design Theme--}}

<script src="{!! asset('js/ripples.min.js') !!}"></script>

<script src="{!! asset('js/material.min.js') !!}"></script>

<script type="text/javascript" src="js/toastr.js"></script>

{{--<script>--}}
    {{--$(document).ready(function(){--}}
        {{--$.material.init();--}}
    {{--});--}}
{{--</script>--}}

</body>
</html>

**Comment by nilsenj** [comment]
in your view place _{!! Toastr::render() !!}_ just before body closing tag

**Comment by nilsenj** [comment]
and don't forget to include  {!! Toastr::clear() !!} after render .

**Comment by nilsenj** [comment]
and in composer.json place the module in require-dev section

**Comment by nilsenj** [comment]
for example 

```
"require-dev": {
    "fzaninotto/faker": "~1.4",
    "mockery/mockery": "0.9.*",
    "phpunit/phpunit": "~4.0",
    "phpspec/phpspec": "~2.1",
    "doctrine/dbal": "^2.6@dev",
    "potsky/laravel-localization-helpers" : "~1.1",
    "nilsenj/toastr-5.1-laravel":"dev-master"
}
```

**Comment by petrovitch** [comment]
Thanks!  The placement of the render tag is important.  Placing it anywhere
other than the line above </body> did not work.

By the way, I did not change the composer tab from required to required-dev
because as I understand it the required-dev tag is just for development,
like phpunit, etc.  I want it to work in production too.

**Comment by nilsenj** [comment]
by the way I've tested copied toastr package into require section and it still works fine!
really thanks a lot! GL

**Comment by petrovitch** [comment]
Ivan,

This probably has nothing to do with your class, but something I don't
understand about Laravel.  I'm using the same code in a class/index and it
works fine, but using the same code in class/show or class/detail it says
Toastr is undefined in the console.  All of these methods are within the
same class.  I even stripped the content of the views to show only the
master layout.

$results = DB::select( DB::raw("SELECT SUM(amount) AS balance FROM gltrns") );
$balance = $results[0]->balance;
if ($balance != 0){
    Toastr::warning('Your journal is out of balance by $' .
number_format($balance,2), 'Validation' );
}

**Comment by petrovitch** [comment]
Found it.  In the master layout I was using an absolute address for the
css/js files.  It looks like when it calls a page from another controller
it does not recognize the path, i.e.

from class/index
css/toastr.css

but from class/show it's looking for this:
method/css/toastr.css

so I had to change the master to

<link rel="stylesheet" type="text/css" href="{!! asset('css/toastr.css') !!}">

and it works fine.
